contracts/ceros/provider/BaseLpTokenProvider.sol

```diff
@@ -0,0 +1,265 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../interfaces/IVault.sol";
+import "../interfaces/IDex.sol";
+import "../interfaces/IDao.sol";
+import "../interfaces/IHelioProviderV2.sol";
+import "../interfaces/IBNBStakingPool.sol";
+import "../interfaces/ICertToken.sol";
+import "../interfaces/IHelioTokenProvider.sol";
+import "../../masterVault/interfaces/IMasterVault.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+
+abstract contract BaseLpTokenProvider is IHelioTokenProvider,
+    AccessControlUpgradeable,
+    PausableUpgradeable,
+    ReentrancyGuardUpgradeable,
+    UUPSUpgradeable
+{
+    using SafeERC20 for IERC20;
+
+    uint128 public constant RATE_DENOMINATOR = 1e18;
+    // manager role
+    bytes32 public constant MANAGER = keccak256("MANAGER");
+    // pause role
+    bytes32 public constant PAUSER = keccak256("PAUSER");
+    // proxy role
+    bytes32 public constant PROXY = keccak256("PROXY");
+
+    /**
+     * Variables
+     */
+    // Tokens
+    address public certToken; // original token, e.g FDUSD
```

Comment:
Better to rename `certToken` to `ceToken` here?

Since certToken here represents the original token, for BNB it's ceABNBc, for ETH it's cewBETH